pkg/source/chirpstack/config/config.go

```diff
@@ -120,8 +121,8 @@ func (c *Config) Initialize() error {
 }
 
 func (c *Config) dialGRPC(opts ...grpc.DialOption) error {
-	if c.insecure && c.caPath == "" {
-		opts = append(opts, grpc.WithInsecure())
+	if c.insecure || c.caPath == "" {
```

Comment:
FWIW I think this should only evaluate `c.insecure`.

If a CA path is not given, it should probably just use system roots and still use TLS